src/main/java/org/jenkinsci/plugins/github_branch_source/ApiRateLimitChecker.java

```diff
@@ -0,0 +1,142 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Util;
+import hudson.model.TaskListener;
+import org.kohsuke.github.GHRateLimit;
+import org.kohsuke.github.GitHub;
+
+import java.io.IOException;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+public enum ApiRateLimitChecker {
+    ThrottleForNormalize(Messages.ApiRateLimitChecker_ThrottleForNormalize()) {
```

Comment:
Cloud you add comments before each option to explain what they do?

Reply:
Add comments from `Messages.properties`.
I can’t write a more detailed description, due my terrible English.

Reply:
This is just fine.